src/python/grpcio/grpc/_cython/_cygrpc/credentials.pyx.pxi

```diff
@@ -137,7 +137,7 @@ cdef void plugin_get_metadata(
   cy_context.context = context
   self.plugin_callback(cy_context, python_callback)
 
-cdef void plugin_destroy_c_plugin_state(void *state):
+cdef void plugin_destroy_c_plugin_state(void *state) with gil:
```

Comment:
I'm not sure about Cython's behavior, but it seems like manipulating python objects in c extensions requires the gil

https://docs.python.org/2/c-api/init.html#thread-state-and-the-global-interpreter-lock

The error I was seeing was a consistent segfault on the first execution of the interop client in a fresh docker instance, and this change fixed it.